models/DataObject/ClassDefinition/Data/Multiselect.php

```diff
@@ -279,6 +279,46 @@ public function checkValidity(mixed $data, bool $omitMandatoryCheck = false, arr
         if (!is_array($data) && !empty($data)) {
             throw new Model\Element\ValidationException("Invalid multiselect data on field [ {$this->getName()} ]");
         }
+
+        if (is_array($data)) {
+            // Ensure options providers are resolved
+            if ($this->getOptions() === null) {
+                $this->enrichFieldDefinition($params['context'] ?? []);
+            }
+
+            foreach ($data as $value) {
+                if (!$this->isValidOption($value)) {
+                    throw new Model\Element\ValidationException(
+                        sprintf("Invalid multiselect option '%s' on field [ %s ]", $value, $this->getName())
+                    );
+                }
+            }
+        }
+    }
+
+    /**
+     * Validates if the provided data matches any of the available options.
+     *
+     */
+    private function isValidOption(mixed $data): bool
+    {
+        // If no options are defined yet, skip the validation
+        if (!$this->getOptions()) {
+            return true;
+        }
+
+        $matches = array_filter(
+            $this->getOptions(),
+            function (array $option) use ($data) {
+                if (!array_key_exists('value', $option)) {
+                    return false;
+                }
+
+                return $option['value'] == $data;
+            }
+        );
+
+        return count($matches) > 0;
```

Comment:
```suggestion
            // If no options are defined yet, skip the validation
            if (!$this->getOptions()) {
                return;
            }

            foreach ($data as $value) {
                if (!$this->isValidOption($value)) {
                    throw new Model\Element\ValidationException(
                        sprintf("Invalid multiselect option '%s' on field [ %s ]", $value, $this->getName())
                    );
                }
            }
        }
    }

    /**
     * Validates if the provided data matches any of the available options.
     *
     */
    private function isValidOption(mixed $data): bool
    {
          $options = $this->getOptions();
          foreach ($options as $option) {
              if (($option['value'] ?? null) == $data) {
                  return true;
              }
          }
      
          return false;
```
So far LGTM, just wanted to add an input to move the validation skip outside the `foreach` loop and to simplify the valid option check. wdyt?